examples/arm/run.sh

```diff
@@ -172,4 +172,8 @@ for i in "${!test_model[@]}"; do
     run_fvp executor_runner.elf
 done
 
+# Run the TOSA Reference_Model e2e tests
+cd $et_root_dir
```

Comment:
These are effectively different lowering and execution paths, and since they are in the same script with `-e` if something fails before this, then this won't run. Do you want to either selectively apply `-e` or split this into running different individual scripts so we can get more complete signal when something fails?

Reply:
I agree we can split into separate files (`run.sh` and `run_tosa_e2e.sh`) which will be much cleaner. If this sounds good, I will push the changes. 

Another question is: do i need to do anything else to trigger the running of the new shell script `run_tosa_e2e.sh` ?

Reply:
hopefully we can add another backend under https://github.com/pytorch/executorch/blob/c44dadf2da17a160768b5aac2e5450d2c835616f/.github/workflows/trunk.yml#L140
Or if preferred from the CI flow we can add it as a script.

I think we should restructure this a bit; I view things as:
  `examples/arm/` are typically user run scripts and tutorials
  `backends/arm/test/*` are unit tests and test scripts

We should add `backends/arm/test/test.sh which tests both the example (as we don't want it to regress either) and the TOSA AoT flow.

Reply:
Agree, we should have these scripts under backends/arm.

Reply:
Thanks Jerry, changes look sensible to me. being nit-picky having the same name for the invocation script and the .py would be cleaner? might be easier if we pattern match it in automation for some reason later too.

Don't think it's worth doing now, but we might want to rename slightly when we have more test flows, as we have test_tosa.py (unit tests which test edge to tosa) and arm_tosa_reference.py which captures the intermediate compiler output (tosa fb, usually a wire format in AoT) and runs it on the reference.  so being consistent as we add more and looking to something like:

for the AoT compiler flows (by the backend options (which runtime target) that are passed):
unit_tosa
unit_vela
unit_<other/many>

test_tosa_reference
test_u55 (e.g. like the examples/arm/ flow but for when we have substantial correctness testing of a u55/65 backend).

Now it's down to how the CI team would prefer to integrate?

Reply:
Yeah let me ping Huy. 

Also I am ok with the change overall besides names bikeshedding which we should converge as we develop more.